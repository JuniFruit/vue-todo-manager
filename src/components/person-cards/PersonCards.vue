<template>
    <v-layout row>
        <v-flex xs12 sm6 md4 lg3 v-for="person in team" :key="person.name">
            <v-card class="text-xs text-center ma-3 pa-4">
                <v-responsive>
                    <v-avatar size="avatarSize" color="gray lighten-2">
                        <img :src="person.avatar" alt="profile avatar">
                    </v-avatar>
                </v-responsive>
                <v-card-text>
                    <div class="subheading">{{ person.name }}</div>
                    <div class="grey--text">{{ person.role }}</div>
                </v-card-text>
                <v-card-actions class="justify-center">
                    <v-btn color="gray" @click="onMessage(person.name)" class="grey--text" :ripple="false" depressed>
                        <v-icon small left>mdi-message</v-icon>
                        <span>Message</span>
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-flex>
    </v-layout>
</template>

<script lang="ts">
import { mutations } from '@/store/store';
import { PropType } from 'vue';
import { IPersonInfo } from './Cards.interface';
export default {
    props: {
        team: {
            required: true,
            type: Array as PropType<IPersonInfo[]>
        }
    },
    data() {
        return {

        }
    },
    methods: {
        onMessage: (person: string) => {
            mutations.setToast(`Sorry, ${person} is busy right now. Try again later`, 'error')
        }
    }
}

</script>
