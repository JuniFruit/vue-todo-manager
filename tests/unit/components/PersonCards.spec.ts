import PersonCards from "@/components/person-cards/PersonCards.vue";
import { Wrapper } from "@vue/test-utils";
import { createCustomWrapper } from "../utils/general";

describe("PersonCards.vue module", () => {
  let wrapper: Wrapper<PersonCards>;
  const users = [
    {
      name: "John Doe",
      role: "tester",
      avatar: "//",
    },
    {
      name: "John Doe2",
      role: "tester2",
      avatar: "//",
    },
  ];

  beforeEach(() => {
    wrapper = createCustomWrapper<PersonCards>(PersonCards, {
      propsData: {
        team: users,
      },
    });
  });

  afterEach(() => {
    wrapper.vm.$destroy();
  });

  it("renders items in a list if it's not empty", () => {
    expect(wrapper.text()).toMatch(users[0].name);
    expect(wrapper.text()).toMatch(users[0].role);
  });
  it("renders vuetify cards", () => {
    const card = wrapper.find(".v-card");
    expect(card).toMatchSnapshot();
  });

  it("renders nothing if it's empty", () => {
    expect(expect(wrapper.text()).toMatch(users[0].name)).toBeFalsy();
  });
});
