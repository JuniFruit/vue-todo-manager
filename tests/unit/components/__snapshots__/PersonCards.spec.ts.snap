// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCards.vue module renders vuetify cards 1`] = `
<div class="text-xs text-center ma-3 pa-4 v-card v-sheet theme--light">
  <div class="v-responsive">
    <div class="v-responsive__content">
      <div class="v-avatar gray lighten-2" style="min-width: avatarSize;"><img src="//" alt="profile avatar"></div>
    </div>
  </div>
  <div class="v-card__text">
    <div class="subheading">John Doe</div>
    <div class="grey--text">tester</div>
  </div>
  <div class="v-card__actions justify-center"><button type="button" class="grey--text v-btn v-btn--has-bg theme--light v-size--default gray"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-message theme--light" style="font-size: 16px;"></i> <span>Message</span></span></button></div>
</div>
`;
